components/scream/src/diagnostics/tests/potential_temperature_test.cpp

```diff
@@ -0,0 +1,277 @@
+  }
+  static bool approx_equal (const ScalarT lhs, const ScalarT rhs,
+                            const int k, const ScalarT tol) {
+    using std::abs;
+    return not ( k<NumLevels && abs(lhs-rhs)>=tol );
+  }
+  static bool approx_equal (const ScalarT computed, const ScalarT expected, const ScalarT tol) {
+    using std::abs;
+    return abs(computed-expected)/abs(expected) < tol;
+  }
+};
+
+template<typename T, int N, int NumLevels>
+struct ChecksHelpers<ekat::Pack<T,N>,NumLevels> {
+  using ScalarT = ekat::Pack<T,N>;
+
+  static bool is_non_negative (const ScalarT& s, const int k) {
+    const auto range = ekat::range<ScalarT>(k*N);
+    const auto range_mask = range < NumLevels;
+    return ( range_mask && (s<0 || isnan(s) ) ).none();
+  }
+  static bool equal (const ScalarT& lhs, const ScalarT& rhs) {
+    return (lhs==rhs).all();
+  }
+  static bool approx_equal (const ScalarT& lhs, const ScalarT& rhs,
+                            const int k, const T tol) {
+    const auto range = ekat::range<ScalarT>(k*N);
+    const auto range_mask = range < NumLevels;
+    return (range_mask && abs(lhs-rhs)>=tol).none();
+  }
+  static bool approx_equal (const ScalarT& computed, const ScalarT& expected, const T tol) {
+    return (abs(computed-expected)/abs(expected) < tol).all();
+  }
+};
+
+// Helper function. Create Mirror View and Deep-Copy (CMVDC)
+template<typename ViewT>
+auto cmvdc (const ViewT& v_d) -> typename ViewT::HostMirror {
+  auto v_h = Kokkos::create_mirror_view(v_d);
+  Kokkos::deep_copy(v_h,v_d);
+  return v_h;
+}
+
+//-----------------------------------------------------------------------------------------------//
+template<typename ScalarT, typename DeviceT>
+void run(std::mt19937_64& engine)
+{
+  using STraits    = ekat::ScalarTraits<ScalarT>;
+  using RealType   = typename STraits::scalar_type;
+
+  using PF         = scream::PhysicsFunctions<DeviceT>;
+  using PC         = scream::physics::Constants<RealType>;
+
+  using KT         = ekat::KokkosTypes<DeviceT>;
+  using ExecSpace  = typename KT::ExeSpace;
+  using TeamPolicy = typename KT::TeamPolicy;
+  using MemberType = typename KT::MemberType;
+  using view_1d    = typename KT::template view_1d<ScalarT>;
+  using rview_1d   = typename KT::template view_1d<RealType>;
+
+  static constexpr auto test_tol = PC::macheps*1e3;
+
+  constexpr int pack_size = sizeof(ScalarT) / sizeof(RealType);
+  using pack_info = ekat::PackInfo<pack_size>;
+
+  constexpr int num_levs = 32; // Number of levels to use for tests.
+  const     int num_mid_packs = pack_info::num_packs(num_levs);
+  const     int num_int_packs = pack_info::num_packs(num_levs+1);
+
+  using Check = ChecksHelpers<ScalarT,num_levs>;
+
+
+  // A world comm
+  ekat::Comm comm(MPI_COMM_WORLD);
+
+  // Create a grids manager - single column for these tests
+  auto gm = create_gm(comm,1,num_levs);
+
+  // Kokkos Policy
+  TeamPolicy policy(ekat::ExeSpaceUtils<ExecSpace>::get_default_team_policy(1, 1));
```

Comment:
For such a trivial policy, you don't need to bring in ExeSpaceUtils (and include its header). You can just do
```
TeamPolicy policy(1,1);
```